classes/semver.js

```diff
@@ -246,9 +246,12 @@ class SemVer {
         break
       // This probably shouldn't be used publicly.
       // 1.0.0 'pre' would become 1.0.0-0 which is the wrong direction.
-      case 'pre':
+      case 'pre': {
+        identifierBase = identifierBase === 'false' ? false : identifierBase
```

Comment:
The cli is the only time this is going to be a string `'false'` and it is the cli that should cast this to `0`, `1`, or `false`.